src/BrowserProtocol.js

```diff
@@ -0,0 +1,94 @@
+import off from 'dom-helpers/events/off';
+import on from 'dom-helpers/events/on';
+import invariant from 'invariant';
+
+import createPath from './utils/createPath';
+
+export default class BrowserProtocol {
```

Comment:
The user does. The interop looks like:

```js
const history = createHistory(
  new BrowserProtocol({ basename: '/active-links' }),
  queryMiddleware,
);
```

These need to actually provide an imperative interface for doing stuff. The last middleware is `createHistoryMiddleware`.